### PR TITLE
remove linkerd error body if clearContext is set

### PR DESCRIPTION
**Pull request by fantayeneh**
Addresses #1163

### CODE REVIEW

**Comment by adleong** [comment]
Thanks @fantayeneh, this is great!  We're prepping the 1.0.0 release right now but we'll get this merged soon after the release.

File: linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/LinkerdHeaders.scala [hunk 1]
```diff
@@ -449,7 +449,14 @@ object Headers {
           req.headerMap -= k
         }
       }
-      service(req)
+      service(req).map { rsp =>
+        rsp.headerMap.get(Err.Key) match {
+          case Some(_) =>
+            rsp.clearContent()
```
Comment (adleong) [inline linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/LinkerdHeaders.scala]: We could also potentially strip the `l5d-err`, `l5d-success-class`, and `l5d-retryable` headers off the response here.  What do you think?

File: namer/fs/src/test/scala/io/buoyant/namer/fs/FsTest.scala [hunk 2]
```diff
@@ -87,18 +91,13 @@ class FsTest extends FunSuite with NamerTestUtil {
       val fs = mapper.readValue[NamerConfig](yaml)
       val namer = fs.mk(Stack.Params.empty)
 
-      val bound = lookupBound(namer, path.drop(fs.prefix.size))
-      assert(bound.size == 1)
-      bound.head.addr.sample() match {
-        case Addr.Bound(addrs, _) =>
-          assert(addrs == Set(
+      eventually {
```
Comment (adleong) [inline namer/fs/src/test/scala/io/buoyant/namer/fs/FsTest.scala]: Does this fix https://github.com/linkerd/linkerd/issues/1203 ?

File: linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/LinkerdHeaders.scala (hunk 1 above)
Comment (fantayeneh) [inline linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/LinkerdHeaders.scala]: Yes make sense.

File: namer/fs/src/test/scala/io/buoyant/namer/fs/FsTest.scala (hunk 2 above)
Comment (fantayeneh) [inline namer/fs/src/test/scala/io/buoyant/namer/fs/FsTest.scala]: I think so. :)

**Comment by adleong** [comment]
Would you mind squashing your commits and formatting the commit message as described in https://github.com/linkerd/linkerd/blob/master/CONTRIBUTING.md?

Thanks!

**Comment by fantayeneh** [comment]
Will create a new one

**Pull request closed without merging**